### ISSUE TITLE
[BUG/ISSUE] Seg fault when dry deposition turned off

### ISSUE THREAD

**Issue by lizziel** [geoschem/geos-chem#205]
In GEOS-Chem Classic 12.6.3, turning off dry deposition results in a segmentation fault when running the standard simulation. It fails during diagnostics write with this meesage:

```
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source             
libifcoremt.so.5   00002AE646BB2216  for__signal_handl     Unknown  Unknown
libpthread-2.17.s  00002AE648C165D0  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F5965  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F56A4  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F72E6  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F75E5  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F7507  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F74BF  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F99B6  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F972D  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486EE823  Unknown               Unknown  Unknown
libiomp5.so        00002AE6486F116E  Unknown               Unknown  Unknown
libifcoremt.so.5   00002AE646BFAA80  for_deallocate        Unknown  Unknown
geos               0000000000C38E37  state_diag_mod_mp        6586  state_diag_mod.F90
geos               00000000004B0221  MAIN__                   2415  main.F
geos               000000000040385E  Unknown               Unknown  Unknown
libc-2.17.so       00002AE6490493D5  __libc_start_main     Unknown  Unknown
geos               0000000000403769  Unknown               Unknown  Unknown
```

I am investigating the problem. It may be in versions prior to 12.6.3 as well.

**Comment by yantosca** [comment]
Does this happen for the benchmark run or for standard?  Wonder if this may be related to the ConcAboveSfc diagnostic.

**Comment by lizziel** [comment]
I only have state_met and speciesconc diagnostics turned on. I'll probably find a fix pretty quickly since I get more info with debug flags on. Stay tuned.

**Comment by lizziel** [comment]
The issue is our restart variable Chem_DryDepNitrogen which is now in HISTORY.rc. Since it contains the substring 'DryDep', and there is a search for substring 'DryDep' in the diagnostic list as criteria to register dry deposition diagnostics, the dry deposition diagnostics are allocated when they should not be. This would be fine (no error) except that the 3rd dimension size is nDryDep which is 0 if dry deposition is off.

I'm going to look into our usage of subroutine check_diaglist to see if we really need it to look for a substring anywhere in the diagnostic name, which is its behavior now. If yes, I'll add an optional argument for requiring that the starts match, similar to the existing optional argument for exact match. Otherwise, I will update it to always require that the starts of the strings match. Returning true if the substring is found anywhere in the diagnostic name is a dangerous.

This update will go into 12.7.1.

**Comment by lizziel** [comment]
This issue is fixed in commit https://github.com/geoschem/geos-chem/commit/1266c23967a562ea76808cf66d4202bb955c6760 and will be available in 12.7.1.
